scripts/lib/CIME/case/check_input_data.py

```diff
@@ -205,13 +244,57 @@ def check_input_data(case, protocol="svn", address=None, input_data_root=None, d
                             no_files_missing = False
 
                             if (download):
+                                if firstdownload:
+                                    # Get the md5 checksum file
+                                    got_chksum = _download_checksum_file(server, input_data_root)
+                                    firstdownload = False
+                                isdirectory=rel_path.endswith(os.sep)
                                 no_files_missing = _download_if_in_repo(server, input_data_root, rel_path.strip(os.sep),
-                                                               isdirectory=rel_path.endswith(os.sep))
+                                                                        isdirectory=isdirectory)
+                                if got_chksum and no_files_missing and not isdirectory:
+                                    verify_chksum(input_data_root,"inputdata_checksum.dat", rel_path.strip(os.sep))
                         else:
+                            if chksum:
+                                verify_chksum(input_data_root,"inputdata_checksum.dat", rel_path.strip(os.sep))
+                                logger.info("Chksum passed for file {}".format(os.path.join(input_data_root,rel_path)))
                             logging.debug("  Already had input file: '{}'".format(full_path))
 
+
+
                 else:
                     model = os.path.basename(data_list_file).split('.')[0]
                     logging.warning("Model {} no file specified for {}".format(model, description))
 
     return no_files_missing
+
+
+def verify_chksum(input_data_root, checksumfile, filename):
+    if not chksum_hash:
```

Comment:
Similar to my above comment about only having one checksum file for all servers: I think it's a problem that you only have a single chksum_hash variable for all servers. My understanding of the logic here is that you'll read the checksum file for the first server, but then for each subsequent server, you won't reset chksum_hash, but instead will use the variable from the first server. If my understanding is right, then this is a problem.